notebooks/speculative-sampling/speculative-sampling.ipynb

```diff
@@ -88,23 +88,68 @@
+    "You can find OpenVINO optimized FastDraft models can be found in this [collection](https://huggingface.co/collections/OpenVINO/speculative-decoding-draft-models-673f5d944d58b29ba6e94161). You can choose either Phi-3 or Phi-4 to be used for demonstrating FastDraft performance. For Phi-3 we will use [Phi-3-mini-4k-instruct-int4-ov](https://huggingface.co/OpenVINO/Phi-3-mini-4k-instruct-int4-ov) as target model and [Phi-3-mini-FastDraft-50M-int8-ov](https://huggingface.co/OpenVINO/Phi-3-mini-FastDraft-50M-int8-ov) as draft. For Phi-4 we will use [Phi-4-mini-instruct-int4-ov](https://huggingface.co/OpenVINO/Phi-4-mini-instruct-int4-ov) as target model and [Phi-4-mini-FastDraft-120M-int8-ov](https://huggingface.co/OpenVINO/Phi-4-mini-FastDraft-120M-int8-ov) as draft.\n",
     "\n",
     "In case, if you want run own models, you should convert them using [Hugging Face Optimum](https://huggingface.co/docs/optimum/intel/openvino/export) library accelerated by OpenVINO integration. More details about model preparation can be found in [OpenVINO LLM inference guide](https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/llm-inference-native-ov.html#convert-hugging-face-tokenizer-and-model-to-openvino-ir-format)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "61ecfe17",
+   "metadata": {},
+   "source": [
+    "### Select model\n",
+    "[back to top ⬆️](#Table-of-contents:)"
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": null,
+   "id": "fe934261",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "\n",
+    "if not Path(\"notebook_utils.py\").exists():\n",
+    "    r = requests.get(\n",
+    "        url=\"https://raw.githubusercontent.com/openvinotoolkit/openvino_notebooks/latest/utils/notebook_utils.py\",\n",
+    "    )\n",
+    "    open(\"notebook_utils.py\", \"w\").write(r.text)\n",
+    "\n",
+    "from notebook_utils import model_widget\n",
+    "\n",
+    "model = model_widget(default=\"Phi-4\")\n",
+    "\n",
+    "model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ccca9ef6",
+   "metadata": {},
+   "source": [
+    "### Download target and draft models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
    "id": "74bb9f96",
    "metadata": {},
    "outputs": [],
    "source": [
     "from pathlib import Path\n",
     "import huggingface_hub as hf_hub\n",
     "\n",
-    "draft_model_id = \"OpenVINO/Phi-3-mini-FastDraft-50M-int8-ov\"\n",
-    "target_model_id = \"OpenVINO/Phi-3-mini-4k-instruct-int4-ov\"\n",
+    "if model == \"Phi-4\":\n",
```

Comment:
that is probably will not work. You compare widget with string and it is always will be false. model.value should be used

Reply:
Done